### DISCUSSION TITLE
Addenda Types

### DISCUSSION THREAD

**Discussion by junav** [moov-io/ach#1419] [Q&A] [answered]
Does the `ach` library support parsing and extracting data from all types of NACHA addenda records, including Addenda05, Addenda06, Addenda07, and Addenda08, from a NACHA file? 

Thank you

**Comment by wadearnold** [comment]
The project supports generating and parsing all Standard Entry Class (SEC) codes and validating files according to the Nacha standards.

**Reply by junav** [reply]
Thanks for the reply. Do you have an example for checking for those addendas, for example 7?

**Comment by wadearnold** [comment, marked as answer]
I may be missing the question. The SEC code of the entry details defines whether an addenda record is mandatory, if it exists, the addenda record type, and the maximum number of added records possible. 

You may be asking if the library supports addenda type 07. I am not aware of that addenda-type code. 

For example, the SEC codes of ACK, ATX, CCD, CIE, CTX, DNE, ENR, WEB, PPD, and TRX all use an Addenda type code of 05.

02: Used for the POS, MTE, and SHR standard entry classes, which provide terminal location information
05: Used for CCD, CTX, and PPD standard entry classes, which provide special addenda information
98: Used for NOCs
99: Used for returns

**Comment by junav** [comment]
So we wanted to check for each addenda type regardless of SEC code. Is there a method for example like getAddendaCode() to extract and check for every Addenda Type for a transaction?

**Reply by adamdecaf** [reply]
There is no such method available. If you're expecting to get custom addenda records you could try parsing them with a forked Reader. Are you trying to support non-standard Addenda records or 05/10/etc in unexpected places within a file?

**Reply by junav** [reply]
I appreciate the clarification. Exactly, my goal is to ensure all addenda records are accounted for regardless of the SEC code within the file. Looking for a way to catch even addendas in unexpected places. Our use case is solely capturing and storing what was passed in the file. Even when its a mismatch.

**Reply by junav** [reply]
Looking at parsing them with a forked Reader. Could you provide further guidance on how to implement this approach, any best practices or working example?

**Reply by adamdecaf** [reply]
The goal of moov-io/ach is to parse all valid Nacha files, so we lock down the reader to expect what the Nacha rules allow. 

Every valid addenda is captured by this library, so it should be enough for your use-case. If you want to fork the reader you'll want to look at [parseEDAddenda](https://github.com/moov-io/ach/blob/master/reader.go#L393) and [parseAddenda](https://github.com/moov-io/ach/blob/master/reader.go#L546) / parseIATAddenda to handle your additional records.

**Reply by junav** [reply]
Will take look into this. appreciate the guidance thank you.
